### PR TITLE
feat[next] Embedded field remove __array__

### PR DESCRIPTION
**Pull request by havogt**
Add `.asnumpy` to `Field`. Implicit conversion via `__array__` creates a problem, because expression `np.float*field` will return ndarray instead of field, because `np.float`'s multiply operator will `asarray(rhs)`.

### CODE REVIEW

File: src/gt4py/next/utils.py [hunk 1]
```diff
@@ -53,6 +58,37 @@ def __exit__(self, *exc):
...
+        if isinstance(args[0], tuple):
+            assert all(isinstance(arg, tuple) and len(args[0]) == len(arg) for arg in args)
+            return tuple(impl(*arg) for arg in zip(*args))
+
+        return fun(*cast(_P.args, args))
+
+    return impl
+
+
+# TODO(havogt): consider moving to module like `field_utils`
```
Comment (havogt) [inline src/gt4py/next/utils.py:91]: I am introducing this in #1365

File: tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py [hunk 2]
```diff
@@ -1119,13 +1122,6 @@ def _invalid_unpack() -> tuple[int32, float64, int32]:
 
 
 def test_constant_closure_vars(cartesian_case):
-    if cartesian_case.backend is None:
```
Comment (havogt) [inline tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py:1122]: this is why we are doing the PR

File: src/gt4py/next/utils.py [hunk 3]
```diff
@@ -53,6 +58,37 @@ def __exit__(self, *exc):
...
+        ((5, 7), 9)
+    """
+
+    @functools.wraps(fun)
+    def impl(*args: Any | tuple[Any | tuple, ...]) -> _R | tuple[_R | tuple, ...]:
+        if isinstance(args[0], tuple):
+            assert all(isinstance(arg, tuple) and len(args[0]) == len(arg) for arg in args)
+            return tuple(impl(*arg) for arg in zip(*args))
+
+        return fun(*cast(_P.args, args))
```
Comment (havogt) [inline src/gt4py/next/utils.py:86]:
```suggestion
        return fun(*cast(_P.args, args)) # mypy doesn't understand that `args` at this point is of type `_P.args`
```

File: src/gt4py/next/common.py [hunk 4]
```diff
@@ -464,6 +464,9 @@ def dtype(self) -> core_defs.DType[core_defs.ScalarT]:
     def ndarray(self) -> core_defs.NDArrayObject:
         ...
 
+    def asnumpy(self) -> np.ndarray:
+        return np.asarray(self.ndarray)
```
Comment (egparedes) [inline src/gt4py/next/common.py:468]: Should this be default implementation or just leave it blank?

File: src/gt4py/next/utils.py [hunk 5]
```diff
@@ -53,6 +58,37 @@ def __exit__(self, *exc):
...
 
+_P = ParamSpec("_P")
+_R = TypeVar("_R")
+
 
 def is_tuple_of(v: Any, t: type[_T]) -> TypeGuard[tuple[_T, ...]]:
     return isinstance(v, tuple) and all(isinstance(e, t) for e in v)
+
+
+def apply_to_tuple_elements(fun: Callable[_P, _R]) -> Callable[..., _R | tuple[_R | tuple, ...]]:
```
[the remark this answers is not in the document]
Comment (egparedes) [reply src/gt4py/next/utils.py:69]: Either deep_map or tree_map (jax calls `tree_map` to something similar)

**Review by egparedes — APPROVED** [review]
A couple of optional suggestions/comments but otherwise it looks good to me.

File: src/gt4py/next/common.py (hunk 4 above)
Comment (havogt) [reply src/gt4py/next/common.py:468]: forgot to remove...

File: src/gt4py/next/utils.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (havogt) [reply src/gt4py/next/utils.py:69]: I'll take tree_map then

**Pull request merged**